### PR TITLE
[PRAY-3193] Fix audio-related crashes

### PR DESCRIPTION
**Pull request by asherepenko**
https://prayinc.atlassian.net/browse/PRAY-3193

- [x] `java.lang.IllegalArgumentException` in BasePlayback.kt - handle unregisterReceiver() exception
- [x] `android.app.RemoteServiceException` because of `Context.startForegroundService() did not then call Service.startForeground()`
- [x] `java.lang.IllegalStateException` in RxMusicPlayer.kt - always launch `MediaService` as foreground service
- [x] https://prayinc.atlassian.net/browse/PRAY-2521 - fix audio focus handling

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -11,12 +11,13 @@ android {
         targetSdkVersion rootProject.targetSdkVersion
         versionCode rootProject.versionCode
         versionName rootProject.versionName
+        multiDexEnabled true
```
Comment (davidzima-pray) [inline app/build.gradle:14]: Do we need this?

File: rxmusicplayer/src/main/java/com/orfium/rx/musicplayer/playback/BasePlayback.kt [hunk 2]
```diff
@@ -67,37 +123,41 @@ internal abstract class BasePlayback(
...
         unregisterWifiLock()
         unregisterNoiseReceiver()
-        stopPlayer()
-        invalidateCurrent()
-    }
-
-    override fun onAudioFocusChange(focusChange: Int) {
-        when (focusChange) {
-            AudioManager.AUDIOFOCUS_GAIN -> {
-                play(currentMedia)
```
Comment (davidzima-pray) [inline rxmusicplayer/src/main/java/com/orfium/rx/musicplayer/playback/BasePlayback.kt:80]: Ah... there's the culprit of the bug I found! 🐛

**Review by davidzima-pray — APPROVED** [review]

File: app/build.gradle (hunk 1 above)
Comment (asherepenko) [reply app/build.gradle:14]: Yeah, you cannot launch sample app without it :(

File: rxmusicplayer/src/main/java/com/orfium/rx/musicplayer/notification/NotificationManager.kt [hunk 3]
```diff
@@ -157,39 +185,13 @@ internal class NotificationManager(
...
-                }
-
-                override fun onLoadFailed(errorDrawable: Drawable?) {
-                    showNotification(builder, null)
-                }
-            })
-    }
-
     private fun showNotification(builder: NotificationCompat.Builder, bitmap: Bitmap?) {
-        if (media == null || state == PlaybackState.Idle) {
```
Comment (davidzima-pray) [inline rxmusicplayer/src/main/java/com/orfium/rx/musicplayer/notification/NotificationManager.kt:183]: @asherepenko Do you remember why you got rid of this check?

**Pull request merged**